source/cluster_manager/clusterManager.js

```diff
@@ -314,6 +314,9 @@ var runAsSlave = function(topicChannel, manager) {
     var loss_count = 0,
         interval;
 
+    clearRouterKeys(topicChannel,manager);
+    var role = "slave";
```

Comment:
It's better to define the `var role` in `ClusterManager` scope, since it's a state of the whole service rather than`runAsSlave` function.